### DISCUSSION TITLE
Unable to use Neumann BCs in 3D

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#18668] [Q&A General] [answered]
Hi all, I'm having trouble applying Neumann BCs on 3D meshes in MOOSE. I initially thought the problem was with the meshes I used previously, which were created by gmsh, but I tried this other mesh made with cubit (both as .inp files). FunctionDirichletBCs work fine (for BOTH meshes), but Neumann conditions give the following error:

```
*** ERROR ***
/home/richmondodufisan/projects/farm/Concrete/test_input2_3D.i:152: (BCs/load/boundary):
    the following side set ids do not exist on the mesh: 3

    MOOSE distinguishes between "node sets" and "side sets" depending on whether
    you are using "Nodal" or "Integrated" BCs respectively. Node sets corresponding
    to your side sets are constructed for you by default.

    Try setting "Mesh/construct_side_list_from_node_list=true" if you see this error.
    Note: If you are running with adaptivity you should prefer using side sets.
```

Even after adding that extra argument to the Mesh block, I still get the same error. Are there any leads for the possible cause of the error? They're both sent to MOOSE as .inp files. 

I also tried exporting the gmsh mesh as a .vtu mesh. With that one, I got a different error:

```
*** ERROR ***
ERROR: negative Jacobian -5.88862e-07 at point (x,y,z)=( 0.34306, 0.326145, -0.000642792) in element 41303
```

Then I tried using the --mesh-only argument to run the file that I saw in a different discussion. The .inp meshes worked fine and showed the Mesh Information, but the .vtu mesh gave an error:

```
*** ERROR ***
Error: Exodus requires all elements with a given subdomain ID to be the same type.
Can't write both TET4 and QUAD4 in the same block!
```

Which is unexpected, because all elements are Quad elements. Again, I want to stress that all meshes worked completely fine with Dirichlet conditions. Also, the 2D meshes I create with gmsh work with Dirichlet and Neumann conditions. My issue is only with 3D meshes. Here are the mesh files used:

[mesh_files.zip](https://github.com/idaholab/moose/files/7024280/mesh_files.zip)

In case it's necessary the gmsh mesh script is saved as a .geo_unrolled file there too.

**Comment by GiudGiud** [comment, marked as answer]
Hello

The side set error can happen for a number of reasons, usually mispelling between the 'boundary=' field and the boundary name in the mesh. Does the sideset appear when you open the mesh in paraview?

The Jacobian error is likely because you have an inverted or very flat element. I had a look at your mesh in Paraview and some elements have rather high quality metrics. 

Guillaume

**Reply by richmondodufisan** [reply]
Hi,

The "boundary=boundary_name" isn't misspelled, the error only comes after changing "Dirichlet" to "Neumann" in the "type=" field (with Dirichlet it runs fine).

We managed to figure out the problem- we needed to slightly modify the problem to run with MOOSE. When we defined the nodeset as two rows of nodes, for some reason MOOSE had trouble converting to sidesets. But when we used 4 rows then it worked fine. This is with the .inp files created by cubit.

**Reply by GiudGiud** [reply]
ok that's odd. Could you please paste the 2 row and 4 row setup here? It's probably a parsing issue of some sort

**Reply by richmondodufisan** [reply]
[beam.zip](https://github.com/idaholab/moose/files/7034511/beam.zip)

These are the two meshes
